### DISCUSSION TITLE
Feature Request: RDP full authentication flow

### DISCUSSION THREAD

**Discussion by alopezme** [thinkst/opencanary#310] [Feature Requests]
I am testing the rdp module with different usernames. But in all the alerts I get the first username entered even if I cahnge it...

I would like also to grab the password if possible.

**Comment by quinn-thinkst** [comment]
Hi @alopezme,

Thanks for writing in. Unfortunately, this is currently a limitation of the lightweight implementation of the RDP service. As it doesn't perform a full authentication negotiation, it doesn't have easy access to all of the user supplied data. Instead, it relies on a cookie to retrieve the username - which is why you see repeating values even after changing your input.

At this time we don't have an immediate fix for the issue, but we've raised it internally for further feature dev.

Thanks again for raising this, and please do let us know if there's anything else we can help you with.

**Reply by alopezme** [reply]
It's me who thanks you for your time working in this interesting project :-)

**Comment by Neleus** [comment]
Could we implement RDP like this ?
https://github.com/GoSecure/pyrdp
It's working better i think. Also capture passwords and logins.
